setup.py

```diff
@@ -0,0 +1,80 @@
+# Installation script for python
```

Comment:
This file should be removed.

Reply:
I did, so I put it on poetry. The thing is that laboneq which we need for the driver does not support python 3.8 @scarrazza . Should we remove ourselves too or ask them to support it ?

Reply:
Well, we can drop it too...

Reply:
Actually, if it's possible, I would try to support py3.8 as well. Of course, if feasible.

The reason is that there are major distributions still quite used that are distributing py3.8 as default `python3`, e.g. [Ubuntu 20.04](https://packages.ubuntu.com/search?keywords=python3&searchon=names&exact=1&suite=focal).
They are definitely not the latest version, nor the latest LTS (for cases in which LTS exist), but still very wide-spread.

Python 3.8 will reach EOL in October 2024. When we'll get closer to the date, for sure we can consider dropping as a simpler alternative. But we're arguably far from there.

Reply:
I can only think of asking them to support python 3.8. Because, I saw that they supported it with older releases but that would mean downgrading the devices firmware and avoiding any updates which I don't think it will be a good idea.

Reply:
The main changes in py3.9 are related to type hints https://docs.python.org/3.9/whatsnew/3.9.html (and interpreter improvements, but they should not affect Python code, at most C and compiled code, and they hove none).

They are actually using the new type hints, e.g.
https://github.com/zhinst/laboneq/blob/e28d5ef3cc308639ad2c63c63a4792f960fa4947/laboneq/openqasm3/namespace.py#L47
but they are also using the old ones
https://github.com/zhinst/laboneq/blob/796777251f918a5119b6a6e53e1ce86fb1bd8ce5/laboneq/dsl/device/device_setup.py#L314

(it is not only the difference in case, `list` vs `List`, but also things that have the same case but are different objects, `collections.abc.Callable` vs `typing.Callable`, because the main change was the introduction of `__class_getitem__`, following [**PEP 585**](https://peps.python.org/pep-0585/))

If they are willing to, and they share the need to keep supporting those users, it should not take them a tremendous effort to make it again compatible with py3.8.

As a possible alternative, if they don't agree, we can also consider lying in the version range for `laboneq`, and in practice require a separate installation procedure (`pip install laboneq==x.y.z` both in development and production).
It'd be annoying, but not unreasonable, and might be more acceptable than dropping py3.8.

However, if anything else is more annoying, and we know no one using Qibolab and py3.8, even dropping it is reasonable. But, if possible, better to support the official Python version range, so please ask them.

Reply:
Thanks @Jacfomg

Reply:
@AleCandido @scarrazza 

The reason for dropping python support below 3.9 was taht we needed to use some of the new language features in 3.9 - If you need details, I can find out for you.
Is this a problem for you, do you need support with the migration?